waiter/src/waiter/core.clj

```diff
@@ -1369,9 +1369,11 @@
                                 [:tokens channels-update-chan-buffer-size tokens-update-chan-buffer-size
                                  watch-refresh-timeout-ms]]]
                               [:state clock kv-store]]
-                             (let [watch-refresh-timer-chan (au/timer-chan watch-refresh-timeout-ms)]
+                             (let [watch-refresh-timer-chan (au/timer-chan watch-refresh-timeout-ms)
+                                   cid-factory-fn (fn create-watch-cid []
+                                                    (str "token-watch-maintainer" "." (utils/unique-identifier)))]
                                (token-watch/start-token-watch-maintainer
-                                 kv-store clock tokens-update-chan-buffer-size channels-update-chan-buffer-size watch-refresh-timer-chan)))})
+                                 kv-store clock tokens-update-chan-buffer-size channels-update-chan-buffer-size watch-refresh-timer-chan cid-factory-fn)))})
```

Comment:
I don't think passing a `cid-factory-fn` is necessary, users can invoke `utils/unique-identifier` directly.